packages/docusaurus-plugin-content-docs/src/plugin-content-docs.d.ts

```diff
@@ -21,18 +21,22 @@ declare module '@docusaurus/plugin-content-docs-types' {
     permalinkToSidebar: PermalinkToSidebar;
   };
 
-  export type PropSidebarItemLink = {
+  export type PropsSidebarItemBase = {
+    extra?: object;
+  };
+
+  export interface PropSidebarItemLink extends PropsSidebarItemBase {
     type: 'link';
     href: string;
     label: string;
-  };
+  }
 
-  export type PropSidebarItemCategory = {
```

Comment:
Can you avoid this refactor please, and keep the existing type aliases instead of changing to interface?

You can do `export PropSidebarItemCategory = PropsSidebarItemBase  & {...}`